lib/cartopy/feature.py

```diff
@@ -187,6 +198,28 @@ def geometries(self):
 
         return iter(geometries)
 
+    def intersecting_geometries(self, extent):
+        """
+        Returns an iterator of shapely geometries that intersect with
+        the given extent.
+        The extent is assumed to be in the CRS of the feature.
+        If extent is None, the method returns all geometries for this dataset.
+
+        """
+
+        if self.autoscale:
+            self.scale = self._scale_from_extent(extent)
```

Comment:
I would have expected this to be called by the geometries method. I'll leave a bigger comment about this at the end of the review.